ballerina/http_commons.bal

```diff
@@ -58,40 +58,50 @@ isolated function buildRequest(RequestMessage message, string? mediaType) return
         request.setByteStream(message);
     } else if (message is mime:Entity[]) {
         request.setBodyParts(message);
+    } else if message is map<string> {
+        match mediaType {
+            mime:APPLICATION_JSON => {
+                json payload = check processJsonContent(message);
+                request.setJsonPayload(payload);
+            }
+            _ => {
```

Comment:
What happens if user sets the media type as something other than `application/x-www-form-urlencoded` after passing a `map<string>`. According to that logic, we still encode and set as text payload. I think we should check the mediatype is null or `application/x-www-form-urlencoded` first. Then we should reat as `application/x-www-form-urlencoded`. For other media types, we parse them according to the media type or give an error. @shafreenAnfar WDYT?

Reply:
I think same level of improvement can come later. The same logic applies for the other scenarios as well. For instance, if you send `bye[]` along with media type `text/plain`, it works without giving any error. If I am not mistaken. 

I believe, our data binding implementations needs to be changed as https://github.com/ballerina-platform/ballerina-standard-library/issues/2530.  We need to do this soon after SL GA.